stglib/iq.py

```diff
@@ -284,14 +372,362 @@ def cdf_to_nc(cdf_filename):
+
+            else:
+                ds["Vel"] = ds["Vel"].where(ds.SNR > ds.attrs["snr_threshold"])
+                ds["Vel_X_Center"] = ds["Vel_X_Center"].where(
+                    (ds.SNR[:, 0] > ds.attrs["snr_threshold"])
+                    & (ds.SNR[:, 1] > ds.attrs["snr_threshold"])
+                )
+                ds["Vel_Z_Center"] = ds["Vel_Z_Center"].where(
+                    (ds.SNR[:, 0] > ds.attrs["snr_threshold"])
+                    & (ds.SNR[:, 1] > ds.attrs["snr_threshold"])
+                )
+                ds["Vel_X_Left"] = ds["Vel_X_Left"].where(
+                    ds.SNR[:, 2] > ds.attrs["snr_threshold"]
+                )
+                ds["Vel_X_Right"] = ds["Vel_X_Right"].where(
+                    ds.SNR[:, 3] > ds.attrs["snr_threshold"]
+                )
+                ds["Vel_Mean"] = ds["Vel_Mean"].where(
+                    (ds.SNR[:, 0] > ds.attrs["snr_threshold"])
+                    & (ds.SNR[:, 1] > ds.attrs["snr_threshold"])
+                    & (ds.SNR[:, 2] > ds.attrs["snr_threshold"])
+                    & (ds.SNR[:, 3] > ds.attrs["snr_threshold"])
+                )
+
+            histtext = "Filled velocity data using snr threshold of {} for corresponding beam(s).".format(
+                Ptxt
+            )
+
+        for var in ds:
+            if "Vel" in var:
+                ds = utils.insert_note(ds, var, histtext)
+
+        ds = utils.insert_history(ds, histtext)
+    else:
+        print("Did not fill velocity data using snr threshold")
+    return ds
+
+
+def fill_vbper(ds):
+    """
+    Fill adjusted pressure, stage, area, range, profile velocity, and depth data with corresponding vertical beam percent good threshold
+    """
+
+    if "vbper_threshold" in ds.attrs:
+        Ptxt = str(ds.attrs["vbper_threshold"])
+
+        histtext = "Filling P1ac, stage, area, range, and D_3 (depth) data using vertical beam percent good threshold threshold of {}.".format(
+            Ptxt
+        )
+
+        notetxt = "Filled data using vertical beam percent good threshold threshold of {}.".format(
+            Ptxt
+        )
+
+        varlist = {"AdjustedPressure", "Depth", "Stage", "Area", "Range"}
+
+        for k in varlist:
+            ds[k] = ds[k].where(ds.VbPercentGood > ds.attrs["vbper_threshold"])
+
+            ds = utils.insert_note(ds, k, notetxt)
+
+        ds = utils.insert_history(ds, histtext)
+
+    else:
+        print(
+            "Did not fill pressure, stage, area, range, and depth data data using vertical beam percent good threshold"
+        )
+
+    return ds
+
+
+def rename_vars(ds):
+    # set up dict of instrument -> EPIC variable names
+
+    ds["vel1_1277"] = ds["Vel"].sel(velbeam=1)
+    ds["vel2_1278"] = ds["Vel"].sel(velbeam=2)
+    ds["vel3_1279"] = ds["Vel"].sel(velbeam=3)
+    ds["vel4_1280"] = ds["Vel"].sel(velbeam=4)
+
+    newvars = {}
+
+    for var in ds:
+        if "Profile_0" in var:
+            newvars[var] = (
```

Comment:
I'm not sure about this huge chained .replace thing here. Is there a cleaner way to do this?

Reply:
Ok I agree. It's a little excessive. What I did was add all these name changes to the dictionary in rename_vars(ds). This created a large list of names but looks better than a chained .replace(). 

But if preferred, I can think of another way to do this without a large list.